lib/flutter_chips_selector.dart

```diff
@@ -8,67 +8,80 @@ import 'package:flutter/scheduler.dart';
+  BuildContext context,
+  ChipsSelectorState<T?> state,
+  T? data,
+);
 typedef ChipsInputSuggestions<T> = FutureOr<List<T>> Function(String query);
 typedef ParsedItems<T> = FutureOr<List<T>> Function(String query);
 
 class ChipsSelector<T> extends StatefulWidget {
-  ChipsSelector(
-      {Key? key,
-      this.initialValue = const [],
-      required this.chipBuilder,
-      required this.suggestionBuilder,
-      required this.findSuggestions,
-      required this.onChanged,
-      this.parseOnLeaving,
-      this.decoration,
-      this.style,
-      this.autofocus,
-      this.keyboardBrightness = Brightness.light,
-      this.textInputType = TextInputType.text,
-      this.textInputAction = TextInputAction.done,
-      this.underlineColor = Colors.transparent,
-      this.labelColor,
-      FocusNode? currentFocus,
-      FocusNode? nextFocus})
-      : this.current = currentFocus ?? FocusNode(),
-        this.next = nextFocus ?? FocusNode(),
+  ChipsSelector({
+    Key? key,
+    this.initialValue = const [],
+    required this.chipBuilder,
+    required this.suggestionBuilder,
+    required this.findSuggestions,
+    required this.onChanged,
+    this.parseOnLeaving,
+    this.decoration,
+    this.style,
+    this.autofocus,
+    this.keyboardBrightness = Brightness.light,
+    this.textInputType = TextInputType.text,
+    this.textInputAction = TextInputAction.done,
+    this.underlineColor = Colors.transparent,
+    this.labelColor,
+    FocusNode? currentFocus,
+    FocusNode? nextFocus,
+  })  : this.textFieldFocusNode = currentFocus ?? FocusNode(), // TODO need to dispose
+        this.nextFocusNode = nextFocus ?? FocusNode(), // TODO need to dispose
         super(key: key);
 
-  final ChipsBuilder chipBuilder;
-  final ChipsBuilder suggestionBuilder;
+  final ChipsBuilder<T> chipBuilder;
+  final ChipsBuilder<T> suggestionBuilder;
   final ChipsInputSuggestions findSuggestions;
+
+  // TODO we should replace this with set since we enforce uniqueness in selectSuggestion anyways
   final List<T> initialValue;
+
+  // TODO we should rename this to onChangeChips or something like that
   final ValueChanged<List<T>> onChanged;
-  final ParsedItems? parseOnLeaving;
+
+  final ParsedItems<T>? parseOnLeaving;
   final InputDecoration? decoration;
   final TextStyle? style;
   final Color underlineColor;
   final Color? labelColor;
   final TextInputType? textInputType;
   final TextInputAction? textInputAction;
-  final FocusNode current;
-  final FocusNode next;
+  final FocusNode textFieldFocusNode;
+  final FocusNode nextFocusNode;
   final bool? autofocus;
   final Brightness keyboardBrightness;
 
+  @Deprecated("Use textFieldFocusNode instead")
+  FocusNode get current => textFieldFocusNode;
+
+  @Deprecated("Use nextFocusNode instead")
+  FocusNode get next => nextFocusNode;
+
   @override
   State<StatefulWidget> createState() => ChipsSelectorState<T>();
 }
 
-class ChipsSelectorState<T> extends State<ChipsSelector<T?>> {
```

Comment:
I never understood why DartCode autogenerated the generic type on a state as nullable. Are you sure we won't get any problems here?

Reply:
ah seems i missed this one, afaik we should never use nullable generic types, since the `T` may be a nullable type itself

Reply:
Okey Dokey